samples/attested_tls/README.md

```diff
@@ -71,7 +71,55 @@ Note: Both of them can run on the same machine or separate machines.
 
 ## Build and run
 
-To build and run the samples, refer to documentation in the main [README file](../README.md#building-the-samples.md).
+### Linux
+
+#### CMake
+- Use Mbed TLS
+  ```bash
+  mkdir build
+  cd build
+  cmake -DOE_CRYPTO_LIB=mbedtls ..
```

Comment:
Are these instructions really specific to this particular sample, or would they apply to many samples?
Why can't we just put this instruction into the ../README.md file?
All the other CMakeLists.txt files you touched here seem to already say:
```
# set OE_CRYPTO_LIB to either "mbedtls" or "openssl" based on the crypto wrapper to be used.
```
So based on that statement already being in the other samples I would say these instructions are in the wrong file and should be moved to ../README.md.

Reply:
Currently, these instructions are specific to the attested tls sample (the others either we don't support or they don't use crypoto APIs at all).  In this case, it's the text in the CMakeList.txt you mentioned is wrong. However, once all the samples support building with OpenSSL, we should update the text based on your suggestion.

Do you think we should fix the CMakelist.txt for now and revisit later?

Reply:
If you're saying the text in all the other CMakeLists.txt files is wrong then yes we should fix them now and revisit later.   But I recommend the README.md change here be in the same PR as the CMakeLists.txt fixes, and you're already touching the other CMakeLists.txt files in this PR so you might as well do it in this PR.

Reply:
The other possibility is to do the README and CMakeLists.txt changes in a separate PR, and remove all the CMakeLists.txt changes for the other samples from this PR.  E.g., if your point is that the helloworld sample, for example, doesn't depend on the crypto library at all, then it doesn't need to have the line you added:
```
set(OE_CRYPTO_LIB
    mbedtls
    CACHE STRING "Crypto library used by enclaves.")
```
You can just delete the OE_CRYPTO_LIB setting if it's not used, right?

Reply:
Ok, I'll go ahead to update the CMakeLists in this PR.

Reply:
I still think each sample application needs crypto library to be mentioned. Aren't all sample applications need OE crypto wrapper library for some internal functions that are not visible to developers?

Reply:
Even if a sample does not use crypto APIs, we still need to link against either oecryptoopenssl or oecryptombedtls (based on the value of `OE_CRYPTO_LIB`) to fullfil the linker requirements. I would say once eventually we should be able to expose the OE_CRYPTO_LIB to user (i.e.., modify the ../README.md as you suggested)

Reply:
@mingweishih , please correct me if I am wrong, but in the helloworldsample: if we were to replace this line in CMakeLists.txt :     enclave openenclave::oeenclave openenclave::oecrypto${OE_CRYPTO_LIB} with openenclave::oecryptoopenssl, wouldnt that just mean the helloworldsample picks up the oecryptoopenssl library?
1) So it is not that the helloworld cannot work with oecryptoopenssl, it is just that we have shown an example of how it works in the attested_tls sample.
2) So if we just update the central readme to say the above, wouldnt that mean that the helloworld sample etc would work with openssl?

Reply:
1 is correct and 2 is mostly correct with two exceptions: data-sealing and file encryptor samples. 
The alternative is update the common README and mention that these two samples are not supported yet.
I don't have preferences between the two approaches. What do you think? @radhikaj  @dthaler

Reply:
My preference is to say that openssl can be used with arbitrary enclaves in this release (i.e., update the common README and mention that those two samples are not updated yet).   I have no preference on whether that's done in this PR or a separate PR, just that I prefer that it be done for this release.

Reply:
Ok, I'll update them in this PR.